zebra-rpc/src/methods.rs

```diff
@@ -1555,6 +1629,25 @@ impl AddressStrings {
         AddressStrings { addresses }
     }
 
+    /// Creates a new [`AddessStrings`] from a given vector, returns an error if any addresses are incorrect.
+    pub fn new_valid(
+        addresses: Vec<String>,
+    ) -> std::result::Result<AddressStrings, server::error::LegacyCode> {
+        let checked_addresses = addresses
+            .into_iter()
+            .map(|address| {
+                address
+                    .parse::<Address>()
+                    .map(|_| address)
+                    .map_err(|_| server::error::LegacyCode::InvalidAddressOrKey)
+            })
+            .collect::<std::result::Result<Vec<_>, server::error::LegacyCode>>()?;
+
+        Ok(AddressStrings {
+            addresses: checked_addresses,
+        })
+    }
```

Comment:
Nitpick/Optional: This could be simplified a bit

```suggestion
    pub fn new_valid(addresses: Vec<String>) -> Result<AddressStrings> {
        let address_strings = Self { addresses };
        address_strings.clone().valid_addresses()?;
        Ok(address_strings)
    }
```

Reply:
Change made in [a65f847].